### DISCUSSION TITLE
cleanup() is not triggered on elements in <template> when their parent is removed

### DISCUSSION THREAD

**Discussion by murisceman** [alpinejs/alpine#2754] [7. Bugs (archived)]
Demo: https://jsfiddle.net/3pvgtyLz/1/ *- check console output*

### We have:

- A custom Alpine directive `test` with a dummy `cleanup` method.
- A dummy component with the ID `#componentRoot`.
- Four child elements with the `x-test` directives.
- A timeout function that removes the component root element after one second.

### Expected behavior:

- When an element is removed, all child elements with the `x-test` directive are "cleaned" and the `el.textContent` is logged in the console.

### Actual behavior:

- The `console.log` triggers only for elements that aren't nested in a `<template>` element.

**Comment by harrynikolov** [comment]
Bumping this bug ^
